OpenLIFUTransducerTracker/OpenLIFUTransducerTracker.py

```diff
@@ -2061,47 +2061,92 @@ def on_transfer_photocollection_from_android_device_clicked(self, checked:bool):
-            return # The logic is responsible for verifying overwrites with the user
+        if data_type == 'photoscan':
+            # Handle photoscan import from Android scan directory
+            # Expected files: texturedMesh.obj, material_0.png, material.mtl, config.json
+            obj_file = None
+            texture_file = None
+            mtl_file = None
+
+            for file_path in pulled_files:
+                filename = os.path.basename(file_path)
+                if filename.endswith('.obj'):
+                    obj_file = file_path
+                elif filename.endswith('.png') or filename.endswith('.jpg'):
+                    texture_file = file_path
+                elif filename.endswith('.mtl'):
+                    mtl_file = file_path
+
+            if not obj_file or not texture_file:
+                slicer.util.errorDisplay(
+                    text="Error: Photoscan is missing required files (model or texture).",
+                    windowTitle="Import Error"
+                )
+                return
 
-        # Verify that there exist imported files when querying the database
+            # Create photoscan dictionary with appropriate structure
+            photoscan_dict = {
+                "model_abspath": obj_file,
+                "texture_abspath": texture_file,
+                "mtl_abspath": mtl_file,
+                "name": f"Photoscan_{cur_reference_number}",
+                "id": cur_reference_number
+            }
 
-        imported_filepaths = get_cur_db().get_photocollection_absolute_filepaths(
-            subject_id=subject_id,
-            session_id=session_id,
-            reference_number=cur_reference_number,
-        )
+            # Add photoscan to database
+            data_logic.add_photoscan_to_database(subject_id, session_id, photoscan_dict.copy())
+            data_logic.update_photoscans_affiliated_with_loaded_session()
 
-        if not imported_filepaths:
-            slicer.util.errorDisplay(
-                text="Error importing files: No files found or import failed.",
-                windowTitle="Import Error"
+            self.updateInputOptions()
+            self.updateWorkflowControls()
+
+            slicer.util.infoDisplay(
+                text=f"Photoscan '{cur_reference_number}' has been successfully imported from the Android device.",
+                windowTitle="Photoscan Imported"
             )
-            return
 
-        # Below is done twice because session_photocollections stored in the
-        # data parameter node is not the same as those stored in
-        # SlicerOpenLIFUSession and both must be updated
-        if photocollection_dict["reference_number"] not in data_parameter_node.session_photocollections:
-            data_parameter_node.session_photocollections.append(photocollection_dict["reference_number"]) # automatically load as well
-        data_logic.update_photocollections_affiliated_with_loaded_session()
+        else:
```

Comment:
suggest explicit `elif data_type=='photos'` here, ending with an `else:` that raises an error about an unhandled case or unexpected `data_type`.